### PR TITLE
Fix key migration process

### PR DESCRIPTION
**Pull request by mitchellhenke**
While testing https://github.com/18F/identity-idp/pull/8253, I found an issue where it's possible to receive an inconsistent read from a fallback key because there's no stage where we write to both keys. To ensure the migration is compatible with zero-downtime, I've added a couple more configuration keys. I've also updated the README to document the migration process as well as our differences from the upstream repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,23 +108,19 @@ Rails.application.config.session_store :redis_session_store,
 By default the Marshal serializer is used. With Rails 4, you can use JSON as a
 custom serializer:
 
-* `:json` - serialize cookie values with `JSON` (Requires Rails 4+)
-* `:marshal` - serialize cookie values with `Marshal` (Default)
-* `:hybrid` - transparently migrate existing `Marshal` cookie values to `JSON` (Requires Rails 4+)
-* `CustomClass` - You can just pass the constant name of any class that responds to `.load` and `.dump`
+* `:marsha` - serialize cookie values with `Marshal` (Default)
```
Comment (zachmargolis) [inline README.md]: lost a letter in here
```suggestion
* `:marshal` - serialize cookie values with `Marshal` (Default)
```

File: lib/redis-session-store.rb [hunk 2]
```diff
@@ -97,11 +108,13 @@ def find_session(req, sid)
...
-    else
-      data = redis_connection.get(prefixed(sid))
-    end
+    data = if read_primary && !read_fallback
+             redis_connection.get(prefixed(sid))
+           elsif read_primary && read_fallback
+             redis_connection.get(prefixed(sid)) || redis_connection.get(prefixed_fallback(sid))
+           elsif !read_primary && read_fallback
+             redis_connection.get(prefixed_fallback(sid))
+           end
```
Comment (zachmargolis) [inline lib/redis-session-store.rb]: is this intended to be long-term code?

my first thought when looking at it was it's missing a case (2 booleans, 4 total branches), I know that we check/prevent the missing 4th case in the initializer, but maybe if we swapped these booleans for an enum, it might be easier? then we have two configs instead of 4?

```
session_id_read_mode: one of [:private_only, :private_with_public_fallback, :public_only]
session_id_write_mode: (same list)
```

**Review by zachmargolis — APPROVED** [review]
LGTM

File: lib/redis-session-store.rb (hunk 2 above)
Comment (mitchellhenke) [reply lib/redis-session-store.rb]: I wasn't super happy with what I have now, I like that a lot more, will update!

Comment (mitchellhenke) [reply lib/redis-session-store.rb]: I tried this a bit, but I think I'd rather keep the flexibility of enabling/disabling independently. I did rename the configuration to be more explicit for public/private ID rather than the fallback terminology that was kind of hard to follow for me.

**Review by zachmargolis — APPROVED** [review]
LGTM

File: lib/redis-session-store.rb (hunk 2 above)
Comment (zachmargolis) [reply lib/redis-session-store.rb]: sgtm, thanks for considering!
